### PR TITLE
Update visualizer to use stack helper 

### PR DESCRIPTION
**Pull request by naresh-bachwani**
This PR adds support of stack helper in postag

I have made the following changes    
1.      Update the postag visualizer to use stack helper

Screenshot (145)

Still to do:
- [x]     Update class prediction error visualizer     
- [x]     Update feature importance visualizer
- [x]     Add a test for sorted tick labels
- [x]     Update bar stack helper if needed

### CODE REVIEW

File: yellowbrick/draw.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def manual_legend(g, labels, colors, **legend_kwargs):
 
 
 def bar_stack(data, ax=None, labels=None, ticks=None, colors=None, 
-              orientation='vertical', colormap=None, **kwargs):
+              orientation='vertical', colormap=None, legend_kws=None, **kwargs):
```
Comment (bbengfort) [inline yellowbrick/draw.py]: In terms of argument ordering, could we put `orientation='vertical'` after `colormap=None`? I think that groups the  args a bit nicer (also update the order in the docstring).

File: yellowbrick/text/postag.py [hunk 2]
```diff
@@ -325,20 +325,12 @@ def draw(self, **kwargs):
...
-            prev = np.zeros(len(self._pos_tags))
-            for index,element in enumerate(pos_tag_counts):
-                self.ax.bar(xidx, 
-                            element, 
-                            bottom = prev, 
-                            color = colors[index])
-                prev+=element
+            bar_stack(pos_tag_counts, ax=self.ax, labels=list(self.labels_), 
+                      ticks=self._pos_tags, colors=self.colors, 
+                      colormap = self.colormap)
```
Comment (bbengfort) [inline yellowbrick/text/postag.py]: This certainly cleaned up a lot of the duplication! Nice one!

File: tests/test_classifier/test_class_prediction_error.py [hunk 3]
```diff
@@ -55,6 +55,8 @@ def test_integration_class_prediction_error_(self):
         model.fit(X, y)
         visualizer = ClassPredictionError(model, classes=["A", "B", "C", "D"])
         visualizer.score(X, y)
+        visualizer.finalize()
```
[the remark this answers is not in the document]
Comment (ndanielsen) [reply tests/test_classifier/test_class_prediction_error.py]: I don't think the recent testsuite changes are necessarily impacting this yet outside of matplotlib 3.x entering in.

We have a PR almost ready to [contains a fix for this remove label issue](https://github.com/DistrictDataLabs/yellowbrick/pull/823/files#r291244069) that seems to have popped up with matplotlib 3.x.

Outside of that, I'd recommend adjusting the tolerances on these tests as the legend text will vary across platforms. 

The other option is removing the legend via `self.assert_images_similar(visualizer, remove_legend=True)

File: tests/test_text/test_postag.py [hunk 4]
```diff
@@ -198,10 +198,21 @@ def test_frequency_mode(self):
         _, ax = plt.subplots()
         tagged_docs = list(get_tagged_docs(sonnets))
 
-        postag(tagged_docs, ax=ax, frequency=True)
+        viz = postag(tagged_docs, ax=ax, frequency=True)
+        viz.poof()
```
[the remark this answers is not in the document]
Comment (ndanielsen) [reply tests/test_text/test_postag.py]: Agree. On certain operating systems, this will pop up a matplotlib image window and slow down the test suite.

File: yellowbrick/draw.py [hunk 5]
```diff
@@ -120,6 +120,9 @@ def bar_stack(data, ax=None, labels=None, ticks=None, colors=None,
 
     colormap : string or matplotlib cmap
         Specify a colormap for each bar, each row in the stack, or every segment.
+        
+    legend_kws : dict, default: None
+        Additional keyword arguments for the legend components.
```
[the remark this answers is not in the document]
Comment (ndanielsen) [reply yellowbrick/draw.py]: I love this, great idea that adds a lot of customization to the visualizer!

**Comment by ndanielsen** [comment]
@naresh-bachwani just wanted to drop a note that this is well crafted PR!

File: tests/test_classifier/test_class_prediction_error.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (bbengfort) [reply tests/test_classifier/test_class_prediction_error.py]: Thanks, @ndanielsen! The `remove_legend` option is very helpful! 

As for the tolerances, @naresh-bachwani is creating these baseline images on Windows - we might want to create the baseline images on a *nix system just to keep things consistent. Perhaps when we get to the end of the PR I can generate the images and set the tolerances? This may only be required if the Travis tolerances are much higher than Windows.

File: tests/test_classifier/test_class_prediction_error.py [hunk 6]
```diff
@@ -66,7 +68,8 @@ def test_class_prediction_error_quickmethod(self):
 
         clf = LinearSVC(random_state=42)
         g = class_prediction_error(clf, X, y, ax, random_state=42)
-
+        # To fit the legend labels within figure area
+        plt.tight_layout(rect=[0, 0, 0.85, 1])
```
[the remark this answers is not in the document]
Comment (naresh-bachwani) [reply tests/test_classifier/test_class_prediction_error.py]: I will try to find a substitute for this. We can skip this here(if you suggest) by using `remove legends` param in `assert_images_similar`.

File: tests/test_text/test_postag.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (naresh-bachwani) [reply tests/test_text/test_postag.py]: I will correct this!

File: tests/test_text/test_postag.py [hunk 7]
```diff
@@ -198,10 +198,21 @@ def test_frequency_mode(self):
...
+        
+        # Sorted tags i.e predetermined order
+        sorted_tags = ['noun', 'adjective', 'punctuation', 'verb', 'preposition',
+                       'determiner', 'adverb', 'conjunction', 'pronoun', 'wh- word', 
+                       'modal', 'infinitive', 'possessive', 'other', 'symbol', 
+                       'existential', 'digit', 'non-English', 'interjection', 'list']
+        # Extract tick labels from the plot
+        ticks_ax = [tick.get_text() for tick in ax.xaxis.get_ticklabels()]
+        #Assert that ticks are set properly
+        assert ticks_ax==sorted_tags
```
[the remark this answers is not in the document]
Comment (naresh-bachwani) [reply tests/test_text/test_postag.py]: Thanks!

File: yellowbrick/classifier/class_prediction_error.py [hunk 8]
```diff
@@ -167,8 +152,6 @@ def finalize(self, **kwargs):
         cmax = max([sum(predictions) for predictions in self.predictions_])
         self.ax.set_ylim(0, cmax + cmax * 0.1)
 
-        # Put the legend outside of the graph
-        plt.legend(bbox_to_anchor=(1.04, 0.5), loc="center left")
         plt.tight_layout(rect=[0, 0, 0.85, 1])
```
[the remark this answers is not in the document]
Comment (naresh-bachwani) [reply yellowbrick/classifier/class_prediction_error.py]: I will find a substitute (if it's there).

File: yellowbrick/draw.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (naresh-bachwani) [reply yellowbrick/draw.py]: Ok! I will add this param. It will provide contributors with a fair amount of control over the plot.

File: yellowbrick/draw.py (hunk 1 above)
Comment (naresh-bachwani) [reply yellowbrick/draw.py]: Will do so!

File: yellowbrick/text/postag.py (hunk 2 above)
Comment (naresh-bachwani) [reply yellowbrick/text/postag.py]: It was a great idea to get a helper for stacked bar charts!

**Comment by naresh-bachwani** [comment]
> @naresh-bachwani just wanted to drop a note that this is well crafted PR!
Thanks @ndanielsen

File: tests/test_classifier/test_class_prediction_error.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (naresh-bachwani) [reply tests/test_classifier/test_class_prediction_error.py]: Great functionality @ndanielsen!

File: yellowbrick/classifier/class_prediction_error.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (naresh-bachwani) [reply yellowbrick/classifier/class_prediction_error.py]: @bbengfort I tried to find an alternative to plt.tight_layout and came across one possible solution. We can set the axes position manually(using `ax.set_position`). But this may fail for different number subplots.
Also plt.tight_layout works on subplots (i.e. it makes sure that labels of subplots do not intersect). So having it does no harm. It my point of view and I would like to know what you think of this. Also let me know if you have some ax function on your mind that can handle this.

File: tests/test_classifier/test_class_prediction_error.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (bbengfort) [reply tests/test_classifier/test_class_prediction_error.py]: @naresh-bachwani as promised, I've pushed a commit where the baseline images were generated on Unix -- waiting to see if this fixes the Travis tests and breaks the AppVeyor tests now.

File: yellowbrick/classifier/class_prediction_error.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (bbengfort) [reply yellowbrick/classifier/class_prediction_error.py]: So I messed around with this some and I'm not terribly happy with what's happening. For example, the `0.85` box width you specified is very large, if the class labels are `[A, B, C]`, then this is far too much and a box width of `0.95` is sufficient to have the legend on the figure nicely. If the labels are `["hobbies", "books", "cinema"]` -- then you would want `0.85` or maybe even `0.75`. Magic numbers like these and global variables are usually warning signs. 

Ideally, we'd want something like option 4 as described in [this StackOverflow answer](https://stackoverflow.com/questions/4700614/how-to-put-the-legend-out-of-the-plot#answer-43439132) - where the bounding box is automatically computed based on the legend size. However, I did try playing around with it and it's not exactly what we want either. We could try adding dedicated legend axes, but then we go down the road of the problems that we're having in the JointPlot visualizer. 

Bottom line is that this is not the problem we're trying to solve in this PR - so if you could make an issue for it, we'll have to get to it in the future! Thank you for researching it!

**Comment by bbengfort** [comment]
@naresh-bachwani ok, so I've done some updates and experiments with the tests; waiting to see if this passes and then I think we may be good to go with this PR? Make sure you do a pull into your feature branch before making any updates!

File: yellowbrick/classifier/class_prediction_error.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (naresh-bachwani) [reply yellowbrick/classifier/class_prediction_error.py]: Sure, I will make an issue out of it. But I had a question. By changing box width the plot size becomes smaller and that needs to be handled, right?

**Comment by bbengfort** [comment]
@naresh-bachwani I've merged the changes made to the test suite, so I'm hoping the Travis-CI tests should pass; after that it's a matter of adjusting the tolerances for AppVeyor if it fails. I think at that point we can merge this in, unless there are any changes still remaining that you want to make?

File: yellowbrick/classifier/class_prediction_error.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (bbengfort) [reply yellowbrick/classifier/class_prediction_error.py]: That's a good question - I experimented some more, it looks like the figure size is fixed, so changing the box width is more of an aspect ratio thing, the subplot is backing off that percentage and pulling the legend with it. More research required I think.

**Comment by naresh-bachwani** [comment]
I have made my changes. You can merge if tests pass!

**Comment by naresh-bachwani** [comment]
I did not pull the recent changes of the merges. I, therefore, saw images with labels. @bbengfort I need to do a minor change then you can go on to merge the branch(if tests pass :) )

**Comment by naresh-bachwani** [comment]
> so I'm hoping the Travis-CI tests should pass; after that it's a matter of adjusting the tolerances for AppVeyor if it fails. 
@bbengfort the baseline images updated for some of the tests still contain legends and labels in this branch. I suppose that is the reason these tests are failing. I can update them if you want (or you can update them on a Unix system to keep things constant!)

**Comment by bbengfort** [comment]
@naresh-bachwani perfect- if you don't mind I can take it from here! Ping me if I forget to merge this! Travis failed with some postag image failures, I fixed those images. Now to see what appveyor does!
